### PR TITLE
#2261 edit work package template hook

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

- Added new hook
- Created new api input
- New api method for edit work package template
- New api url and passing in payload and id

(Edit work package template is unique in the fact that it takes in the id in the url rather than the payload)

Testing:
Screenshot 2024-04-30 at 9 53 15 PM
Screenshot 2024-04-30 at 9 54 46 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #2261 )

### CODE REVIEW

File: src/frontend/src/hooks/work-packages.hooks.ts [hunk 1]
```diff
@@ -75,6 +77,27 @@ export const useEditWorkPackage = (wbsNum: WbsNumber) => {
   );
 };
 
+/**
+ * Custom React Hook to edit a work package.
```
Comment (Aaryan1203) [inline src/frontend/src/hooks/work-packages.hooks.ts:81]: work package template*

File: src/frontend/src/hooks/work-packages.hooks.ts [hunk 2]
```diff
@@ -75,6 +77,27 @@ export const useEditWorkPackage = (wbsNum: WbsNumber) => {
...
+ *
+ * @returns React-query tility functions exposed by the useMutation hook
```
Comment (Aaryan1203) [inline src/frontend/src/hooks/work-packages.hooks.ts:83]: utility*

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Looks good, left some spelling fixes. Did you test that this works with the endpoint u created?

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Change the name of the pr from everything works to what it actually does

File: src/frontend/src/utils/urls.ts [hunk 3]
```diff
@@ -52,9 +52,20 @@ const workPackages = (queryParams?: { [field: string]: string }) => {
...
 };
+
+const workPackageTemplates = (queryParams?: { [field: string]: string }) => {
+  const url = `${API_URL}/template`;
+  if (!queryParams) return url;
+  return `${url}?${Object.keys(queryParams)
+    .map((param) => `${param}=${queryParams[param]}`)
+    .join('&')}`;
+};
+
```
Comment (Aaryan1203) [inline src/frontend/src/utils/urls.ts:63]: No need for this, i think you can just hardcode the URL

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
left a comment

File: src/frontend/src/utils/urls.ts (hunk 3 above)
Comment (superhvarn) [reply src/frontend/src/utils/urls.ts]: bruh

**Review by Aaryan1203 — APPROVED** [review]
LGTM!

File: src/frontend/src/apis/work-packages.api.ts [hunk 4]
```diff
@@ -20,6 +20,27 @@ export interface WorkPackageApiInputs {
...
+  templateNotes: string;
+  duration: number | undefined;
+  stage?: WorkPackageStage;
+  blockedBy: BlockedByInfo[];
+  expectedActivities: string[];
+  deliverables: string[];
+  workPackageName?: string;
+}
+
+export interface CreateWorkPackageApiInputs extends WorkPackageApiInputs {
```
Comment (Peyton-McKee) [inline src/frontend/src/apis/work-packages.api.ts:34]: what is this? is this being used anywhere?

**Review by Peyton-McKee — APPROVED** [review]
noice

**Pull request merged**